### ISSUE TITLE
Onsite Search Performed

### ISSUE THREAD

**Issue by braxton-butcher-ftr** [further-external/loews-dl-spec-ga4#23]
https://github.com/searchdiscovery/loews-dl-spec-ga4/blob/main/Data%20Layer%20Events/Onsite%20Search%20Performed.md

The new Google dataLayer events should fire in all scenarios where the corresponding appEventData data layer event fires. 

Please reference the [Data Layers Event Mapping](https://docs.google.com/spreadsheets/d/1PDhNOzXI9E7jZ9obejV4owtW3Wtwq66_IaN-CBoHbRs/edit#gid=1910664823) document to see the relationship between the two data layers events.

Reference the [Data Layers Attribute Mapping](https://docs.google.com/spreadsheets/d/1PDhNOzXI9E7jZ9obejV4owtW3Wtwq66_IaN-CBoHbRs/edit#gid=382527185) document to see how the specific attributes of the old appEventData data layer corresponds to the attributes of the new dataLayer data layer. 

Reference the [Location ID Mapping](https://docs.google.com/spreadsheets/d/1PDhNOzXI9E7jZ9obejV4owtW3Wtwq66_IaN-CBoHbRs/edit#gid=1543857253) document to see how to populate ecommerce.items.item_id and event_data.location_id in the new data layer. 
Both of these should contain the corresponding hotel code as found in the aforementioned document.

You may also reference the “Attached Notes” section of the event’s specs to determine where and when to fire.

**Comment by escohen115** [comment]
Please fire the GA4 data layer event (i.e. "datalayer") in all scenarios where the Adobe data layer event (i.e. "appEventData") is firing

**Comment by JH-SDI** [comment]
@dbokadiya @brendenconcannon 
 
Moving back this to "In Progress" this event needs to fire site wide. 

Event fires on Booking Engine but missing properties:



![Image](https://github.com/searchdiscovery/loews-dl-spec-ga4/assets/121498633/0fe7c2b6-3278-4f8b-a9c8-95002ddef055)

  "days_to_start_date": <days_to_start_date>,
        "end_date": "<end_date>",
        "market_code": "<market_code>",
        "number_of_adults": <number_of_adults>,
        "number_of_children": <number_of_children>,
        "rate_code": "<rate_code>",
        "start_date": "<start_date>",

**Comment by estum1** [comment]
It's not clear what the market code data is supposed to be but all other information is pulling in

**Comment by JH-SDI** [comment]
@estum1 @brendenconcannon
 
Moving back this to "To DO" 

Event fires in the correct scenario on the BE:



![Image](https://github.com/searchdiscovery/loews-dl-spec-ga4/assets/121498633/1717c7dd-1b09-457f-8058-7d3cb67135e8)


But it doesn't fire on the main site(we are using testing environment: https://loewshotels-preview.dev.cendynecommerce.com/)

this event needs to fire site wide.

**Comment by estum1** [comment]
Firing for me on the main search bar. 

![Image](https://github.com/searchdiscovery/loews-dl-spec-ga4/assets/40218225/026cc75c-1982-4acb-af34-f366b1f7e6c6)

**Comment by braxton-butcher** [comment]
@estum1 
Event fires in the correct scenario, but some improvements are needed

![Image](https://github.com/searchdiscovery/loews-dl-spec-ga4/assets/113931881/929fada4-9001-4d02-a207-c73230e02535)

days_to_start_date: This should be an integer equalling the number of days between the date the search is performed and the first day of hat search. In this example it would be 0. The data type should be integer

location_id: This should be a string representing the numeric ID of the hotel being searched for. [See this document for full hotel ID mapping](https://docs.google.com/spreadsheets/d/1PDhNOzXI9E7jZ9obejV4owtW3Wtwq66_IaN-CBoHbRs/edit#gid=1543857253)

market_code: If market code is unavailable, this should populate as an empty string rather than null

number_of_adults: This should be an integer, not a string

type: Should be the type of search ("location", "availability", or "room"). This should be equivalent to the parameter found in onsiteSearch.keyword.searchType in the current implementation of appEventData. [See ull mapping of appEventData to the new dataLayer in this document](https://docs.google.com/spreadsheets/d/1PDhNOzXI9E7jZ9obejV4owtW3Wtwq66_IaN-CBoHbRs/edit#gid=382527185)

**Comment by braxton-butcher** [comment]
@estum1 
Event fires in the correct scenario, but some improvements are needed

![Image](https://github.com/searchdiscovery/loews-dl-spec-ga4/assets/113931881/9bf5fb93-f043-4124-98ad-849dd0155025)

days_to_start_date: This should be an integer equaling the number of days between the date the search is performed and the first day of hat search. In this example it would be 0. The data type should be integer, not an empty string

end_date: Please provide the date in format "YYYY-MM-DD". In this example it would be "2023-07-01". Sorry for not calling this out earlier!

location_id: While this is in the correct format, please ensure that it is the correct location ID. I completed a search for Chicago hotel, which has ID "70564". The returned ID was "70569", which is the ID for the New Orleans hotel.

market_code: If market code is unavailable, this should populate as an empty string rather than undefined

start_date: Please provide the date in format "YYYY-MM-DD". In this example it would be "2023-06-30". Sorry for not calling this out earlier!

**Comment by estum1** [comment]
This is updated except the date format - is it critical it be in that format? We would have to do a good bit of updating on our end to get it there.

**Comment by braxton-butcher** [comment]
@estum1 Yes it is very important to be in that format - This is the format required by GA4 and data will not be captured correctly if it is in a different format

**Comment by braxton-butcher** [comment]
@estum1 The date format looks good, the only issue I'm currently seeing is with the value of days_to_start_date


![Image](https://github.com/searchdiscovery/loews-dl-spec-ga4/assets/113931881/2a2b5c42-6047-4d9d-9c2e-5da933993ad1)


Right now, if I search for a booking on the same day, a value of -1 populates when I would expect this value to be 0


![Image](https://github.com/searchdiscovery/loews-dl-spec-ga4/assets/113931881/4a4a218e-06ea-44aa-8031-bf731204d5e3)

The same issue occurs when I try to search for a booking for tomorrow, I expect a value of 1 but a value of 0 populates.

Moving this back to To Do but once this is resolved we should be good to go!

**Comment by braxton-butcher** [comment]
@estum1 
Looks good! Moving to done


![Image](https://github.com/searchdiscovery/loews-dl-spec-ga4/assets/113931881/4c04b746-fd2b-4cc2-99b4-2b05e4c3fc2b)
